devicetypes/smartthings/zwave-siren.src/zwave-siren.groovy

```diff
@@ -189,23 +249,60 @@ def ping() {
 
 def refresh() {
 	log.debug "sending battery refresh command"
-	delayBetween([
-		zwave.basicV1.basicGet().format(),
-		zwave.batteryV1.batteryGet().format()
-	], 2000)
+	def cmds = []
+	cmds << secure(zwave.basicV1.basicGet())
+	cmds << secure(zwave.batteryV1.batteryGet())
+	if (isYale()) {
+		cmds << secure(zwave.alarmV2.alarmGet(zwaveAlarmType: 0x07))
+	}
+	return delayBetween(cmds, 2000)
 }
 
 def parse(String description) {
 	log.debug "parse($description)"
 	def result = null
-	def cmd = zwave.parse(description, [0x20: 1])
-	if (cmd) {
-		result = zwaveEvent(cmd)
+
+	if (description.startsWith("Err")) {
+		if (state.sec) {
+			result = createEvent(descriptionText: description, displayed: false)
+		} else {
+			result = createEvent(
+				descriptionText: "This device failed to complete the network security key exchange. If you are unable to control it via SmartThings, you must remove it from your network and add it again.",
+				eventType: "ALERT",
+				name: "secureInclusion",
+				value: "failed",
+				displayed: true,
+			)
+		}
+	} else {
+		def cmd = zwave.parse(description, [0x20: 1])
+		if (cmd) {
+			result = zwaveEvent(cmd)
+		}
 	}
 	log.debug "Parse returned ${result?.descriptionText}"
 	return result
 }
 
+private secure(physicalgraph.zwave.Command cmd) {
+	if ((zwaveInfo.zw == null && state.sec != 0) || zwaveInfo?.zw?.contains("s")) {
```

Comment:
you don't need the `state.sec` check because that's a legacy structure we have kept for backwards compatibility on old DTHs where security was handled differently.